### ISSUE TITLE
[New Feature] Allow CrossProtocolRedirects to be configured.

### ISSUE THREAD

**Issue by preetjdp** [florent37/Flutter-AssetsAudioPlayer#188]
**The Problem**
The plugin throws the following error if the network URL passed is a proxy URL and performs Cross Protocol Redirects

```
E/ExoPlayerImplInternal(27298): Source error
E/ExoPlayerImplInternal(27298):   com.google.android.exoplayer2.upstream.HttpDataSource$InvalidResponseCodeException: Response code: 302
E/ExoPlayerImplInternal(27298):       at com.google.android.exoplayer2.upstream.DefaultHttpDataSource.open(DefaultHttpDataSource.java:300)
E/ExoPlayerImplInternal(27298):       at com.google.android.exoplayer2.upstream.DefaultDataSource.open(DefaultDataSource.java:177)
E/ExoPlayerImplInternal(27298):       at com.google.android.exoplayer2.upstream.StatsDataSource.open(StatsDataSource.java:83)
E/ExoPlayerImplInternal(27298):       at com.google.android.exoplayer2.source.ProgressiveMediaPeriod$ExtractingLoadable.load(ProgressiveMediaPeriod.java:956)

```

**Describe the feature you'd like**
Add the ability to set the value for `allowCrossProtocolRedirects`

**Current Solution**
As per the links mentioned below I just set `allowCrossProtocolRedirects` to true.

```                                           
val dataSource = DefaultHttpDataSource(userAgent,DefaultHttpDataSource.DEFAULT_CONNECT_TIMEOUT_MILLIS,DefaultHttpDataSource.DEFAULT_READ_TIMEOUT_MILLIS,true,null)
```

A rough implementation can be seen [here](https://github.com/preetjdp/Flutter-AssetsAudioPlayer/blob/Configure-Cross-Protocol-Redirects/android/src/main/kotlin/com/github/florent37/assets_audio_player/playerimplem/PlayerImplemExoPlayer.kt#L73).

**Proposed Solution**
The ideal solution would be to pass a parameter in dart-side which would configure it for android (maybe even IOS, I'm not well versed with IOS redirect spec).

**References**
* https://exoplayer.dev/doc/reference/com/google/android/exoplayer2/upstream/DefaultHttpDataSourceFactory.html
* https://github.com/google/ExoPlayer/issues/1190
* https://exoplayer.dev/troubleshooting.html#why-do-some-streams-fail-with-http-response-code-301-or-302

**Comment by florent37** [comment]
can you try to add this into a pull request ? I'll add it :)

**Comment by florent37** [comment]
I just set it to tru as default, 

```
return ProgressiveMediaSource.Factory(DataSource.Factory {
                    val allowCrossProtocol = true
                    val dataSource = DefaultHttpDataSource(userAgent, DefaultHttpDataSource.DEFAULT_CONNECT_TIMEOUT_MILLIS, DefaultHttpDataSource.DEFAULT_READ_TIMEOUT_MILLIS, allowCrossProtocol, null)
                    networkHeaders?.forEach {
                        it.key?.let { key ->
                            it.value?.let { value ->
                                dataSource.setRequestProperty(key.toString(), value.toString())
                            }
                        }
                    }
                    dataSource;
                }).createMediaSource(uri)
```

what can be wrong ?

**Comment by florent37** [comment]
maybe it can stay like this :) I publish it

**Comment by florent37** [comment]
published in `2.0.3+2`
